content/patterns/layered-zero-trust/lzt-secure-supply-chain.adoc

```diff
@@ -0,0 +1,557 @@
++
+[source,terminal]
+----
+# Request a new access token
+$ export OIDC_TOKEN="$(curl -sk ${OIDC_TOKEN_URL} \
+  --header 'Accept: application/json' \
+  --data-urlencode 'grant_type=password' \
+  --data-urlencode "client_id=${RHTAS_CLIENT_ID}" \
+  --data-urlencode "username=${RHTAS_USER}" \
+  --data-urlencode "password=${RHTAS_USER_PASSWORD}" \
+  --data-urlencode 'scope=openid email profile' \
+  | jq -r .access_token)"
+----
+
+[id="create-signature-bundle"]
+=== Creating a signature bundle
+
+Run the signing command. This generates a `.bundle` file that has the signature and the transparency log entry.
+
+.Procedure
+
+. Get the Fulcio service hostname:
++
+[source,terminal]
+----
+# Get Fulcio service hostname
+$ export FULCIO_URL="https://$(oc get route -n trusted-artifact-signer -l app.kubernetes.io/component=fulcio -o jsonpath='{.items[0].spec.host}')"
+----
+
+. Get the Rekor service hostname:
++
+[source,terminal]
+----
+# Get Rekor service hostname
+$ export REKOR_URL="https://$(oc get route -n trusted-artifact-signer -l app.kubernetes.io/component=rekor-server -o jsonpath='{.items[0].spec.host}')"
+----
+
+. Set the path to the artifact:
++
+[source,terminal]
+----
+$ export ARTIFACT_PATH="target/qtodo-${VERSION}-runner.jar"
+----
+. Sign the artifact:
++
+[source,terminal]
+----
+# Sign the artifact
+$ cosign sign-blob "${ARTIFACT_PATH}" \
+  --fulcio-url="${FULCIO_URL}" \
+  --rekor-url="${REKOR_URL}" \
+  --identity-token "${OIDC_TOKEN}" \
+  --oidc-client-id="${RHTAS_CLIENT_ID}" \
+  --bundle "${ARTIFACT_PATH}.bundle" \
+  --yes
+----
+
+.Verification
+
+To verify the signed artifact, you can use the generated bundle and the artifact itself. Ensure `cosign` still points to the RHTAS TUF root.
+
+* Verify the artifact by using the following command:
++
+[source,terminal]
+----
+# Verify the artifact signature
+$ cosign verify-blob "${ARTIFACT_PATH}" \
+  --rekor-url="${REKOR_URL}" \
+  --bundle "${ARTIFACT_PATH}.bundle" \
+  --certificate-identity-regexp ".*${RHTAS_USER}.*" \
+  --certificate-oidc-issuer-regexp "${OIDC_ISSUER_URL}"
+----
++
+Beyond command-line verification, you can locate the record by using the Rekor search UI in your web browser. You can search records by email address or record index.
++
+You can get the URL for the Rekor Search UI by using the following command:
+[source,terminal]
+----
+$ oc get route -n trusted-artifact-signer -l app.kubernetes.io/component=rekor-ui -o jsonpath='{.items[0].spec.host}'
+----
+
+image::/images/layered-zero-trust/rekor-web-ui.png[Rekor's Search UI]
+
+[id="sign-container-image"]
+== Signing the image
+
+To sign the image, use RHTAS services again. For this step, you must also be logged in to the Quay service, because you write the digital signature to the repository.
+
+To reference the image you intend to sign, you must use its digest rather than its tag. You can obtain the digest of the image by using `skopeo` or by checking the Quay repository.
```

Comment:
```suggestion
To reference the image you intend to sign, you must use its digest rather than its tag. Get the digest of the image by using `skopeo` or by checking the Quay repository.
```